src/main/java/com/yahoo/bullet/storm/drpc/DRPCQueryResultPubscriber.java

```diff
@@ -0,0 +1,143 @@
+/*
+ *  Copyright 2017, Yahoo Inc.
+ *  Licensed under the terms of the Apache License, Version 2.0.
+ *  See the LICENSE file associated with the project for terms.
+ */
+package com.yahoo.bullet.storm.drpc;
+
+import com.yahoo.bullet.BulletConfig;
+import com.yahoo.bullet.RandomPool;
+import com.yahoo.bullet.pubsub.PubSubException;
+import com.yahoo.bullet.pubsub.PubSubMessage;
+import com.yahoo.bullet.pubsub.Publisher;
+import com.yahoo.bullet.pubsub.Subscriber;
+import com.yahoo.bullet.storm.drpc.utils.DRPCError;
+import lombok.AccessLevel;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.asynchttpclient.AsyncHttpClient;
+import org.asynchttpclient.AsyncHttpClientConfig;
+import org.asynchttpclient.DefaultAsyncHttpClient;
+import org.asynchttpclient.DefaultAsyncHttpClientConfig;
+import org.asynchttpclient.Response;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+/**
+ * This class is both a query {@link Publisher} and a result {@link Subscriber}. It uses HTTP to do a request- response
+ * call where the request is the Bullet query and the response is the result from the backend.
+ */
+@Slf4j
+public class DRPCQueryResultPubscriber implements Publisher, Subscriber {
+    private RandomPool<String> urls;
+    private Queue<PubSubMessage> responses;
+    @Setter(AccessLevel.PACKAGE)
+    private AsyncHttpClient client;
+
+    public static final int NO_TIMEOUT = -1;
+    public static final int OK_200 = 200;
+    // PROTOCOL://URL:PORT/DRPC_PATH/DRPC_FUNCTION
+    private static final String URL_TEMPLATE = "%1$s://%2$s:%3$s/%4$s/%5$s";
+
+    /**
+     * Create an instance from a given {@link BulletConfig}. Requires {@link DRPCConfig#DRPC_HTTP_CONNECT_TIMEOUT_MS},
+     * {@link DRPCConfig#DRPC_HTTP_CONNECT_RETRY_LIMIT}, {@link DRPCConfig#DRPC_SERVERS}, {@link DRPCConfig#DRPC_HTTP_PORT},
+     * and {@link DRPCConfig#DRPC_HTTP_PATH} to be set. To be used in PubSub.Context#QUERY_SUBMISSION mode.
+     *
+     * @param config A non-null config that contains the required settings.
+     */
+    public DRPCQueryResultPubscriber(BulletConfig config) {
+        Objects.requireNonNull(config);
+
+        Number connectTimeout = config.getRequiredConfigAs(DRPCConfig.DRPC_HTTP_CONNECT_TIMEOUT_MS, Number.class);
+        Number retryLimit = config.getRequiredConfigAs(DRPCConfig.DRPC_HTTP_CONNECT_RETRY_LIMIT, Number.class);
+        List<String> urls = (List<String>) config.getRequiredConfigAs(DRPCConfig.DRPC_SERVERS, List.class);
```

Comment:
This local var name is the same as line 38 and make it a little bit hard to read. Can you change to a different name?

Reply:
Will do